### PR TITLE
Add is_loading and is_saving values to cereal::{Input,Output}Archive

### PR DESCRIPTION
**Pull request by headupinclouds**
This provides compatibility for asymmetric serialize routines when transitioning from boost serialization and associated archive types.

See: https://github.com/USCiLab/cereal/issues/360

Example:
```
    template <typename Archive>
    void serialize(Archive &ar, const std::uint32_t version)
    {
        std::int16_t tmp;
        if(Archive::is_loading::value)
        {
            ar & tmp; // load 16 bit type from archive
            m_foo = inflate<std::int32_t>(tmp); // convert to 32 bit for processing
        }
        else
        {
            tmp = deflate<std::in16_t>(m_foo); // convert to 16 bit for storage
            ar & tmp; // save 16 bit
        }
    }
```

### CODE REVIEW

File: include/cereal/cereal.hpp [hunk 1]
```diff
@@ -611,6 +625,20 @@ namespace cereal
           a large project from Boost to cereal.  The preferred interface for cereal is using operator(). */
       //! @{
 
+      //! Indicates this archive is intended for loading
+      /*! This ensures compatibility with boost archive types.  If you are transitioning
+          from boost, you can check this value within a member or external serialize function
+          (i.e., Archive::is_loading::value) to enable behavior specific to loading, until 
+          you can transition to split save/load or save_minimal/load_minimal functions */
+.      using is_loading = std::true_type;
```
Comment (erichkeane) [inline include/cereal/cereal.hpp]: Build failed due to the extra '.' at the beginning of this line.

**Comment by erichkeane** [comment]
Also, the ability to use SFINAE is likely better than your example!  Something like:
    template <typename Archive, typename std::enable_if<Archive::is_loading::value,int> =0>
    void serialize(Archive &ar, const std::uint32_t version)

File: include/cereal/cereal.hpp (hunk 1 above)
Comment (headupinclouds) [inline include/cereal/cereal.hpp]: Thanks.  Fixed.

**Comment by headupinclouds** [comment]
> Also, the ability to use SFINAE is likely better than your example!

Nice.

Let me know if you need any follow up.

**Pull request merged**